tracex_project/extraction/forms.py

```diff
@@ -2,52 +2,65 @@
 from django import forms
 
 from extraction.models import PatientJourney
-from tracex.logic.constants import EVENT_TYPES, LOCATIONS, ACTIVITY_KEYS
+from tracex.logic.constants import (
+    EVENT_TYPES,
+    LOCATIONS,
+    ACTIVITY_KEYS,
+    MODULES_OPTIONAL,
+    MODULES_REQUIRED,
+)
 
 
 class BaseEventForm(forms.Form):
     """Base form for event extraction forms."""
 
+    modules_required = forms.MultipleChoiceField(
+        label="Required modules",
+        choices=MODULES_REQUIRED,
+        widget=forms.CheckboxSelectMultiple(),
+        required=True,
+        initial=[module[0] for module in MODULES_REQUIRED],
+        disabled=True,
+    )
+    modules_optional = forms.MultipleChoiceField(
+        label="Select additional modules",
+        choices=MODULES_OPTIONAL,
+        widget=forms.CheckboxSelectMultiple(),
+        required=False,
+        initial=[module[0] for module in MODULES_OPTIONAL],
+        help_text="If modules are deselected, the resulting dataframe will be filled with placeholders!",
+    )
     event_types = forms.MultipleChoiceField(
         label="Select desired event types",
         choices=EVENT_TYPES,
         widget=forms.CheckboxSelectMultiple(),
         required=False,
         initial=[event_type[0] for event_type in EVENT_TYPES],
+        help_text="'N/A' only occurs, if 'Event Type Classifier' is not selected.",
     )
     locations = forms.MultipleChoiceField(
         label="Select desired locations",
         choices=LOCATIONS,
         widget=forms.CheckboxSelectMultiple(),
         required=False,
         initial=[location[0] for location in LOCATIONS],
+        help_text="'N/A' only occurs, if 'Location Extractor' is not selected.",
     )
     activity_key = forms.ChoiceField(
         label="Select activity key for output",
         choices=ACTIVITY_KEYS,
         widget=forms.RadioSelect(),
         required=True,
-        initial=ACTIVITY_KEYS[0][0], # selects the first activity-key in the list, should be event_type
+        initial=ACTIVITY_KEYS[0][
+            0
+        ],  # selects the first activity-key in the list, should be event_type
```

Comment:
suggestion: "# initialize the activity_key with event_type as key" or sth like this